security/pkg/k8s/tokenreview/k8sauthn.go

```diff
@@ -196,44 +234,52 @@ func (authn *K8sSvcAcctAuthn) ValidateK8sJwt(targetToken, jwtPolicy string) ([]s
 		}
 	}
 	if !inServiceAccountGroup {
-		return nil, fmt.Errorf("the token is not a service account")
+		return nil, fmt.Errorf("found the jwt token in cluster %v, however the token is "+
+			"not associated with a service account", clusterID)
 	}
 	// "username" is in the form of system:serviceaccount:{namespace}:{service account name}",
 	// e.g., "username":"system:serviceaccount:default:example-pod-sa"
 	subStrings := strings.Split(tokenReview.Status.User.Username, ":")
 	if len(subStrings) != 4 {
-		return nil, fmt.Errorf("invalid username field in the token review result")
+		return nil, fmt.Errorf("found the jwt token in cluster %v, however found invalid username field "+
+			"in the token review result", clusterID)
 	}
 	namespace := subStrings[2]
 	saName := subStrings[3]
-
 	return []string{namespace, saName}, nil
 }
 
-// isTrustworthyJwt checks if a jwt is a trustworthy jwt type.
-func isTrustworthyJwt(jwt string) (bool, error) {
-	type trustWorthyJwtPayload struct {
-		Aud []string `json:"aud"`
-		Exp int      `json:"exp"`
-	}
-
-	jwtSplit := strings.Split(jwt, ".")
-	if len(jwtSplit) != 3 {
-		return false, fmt.Errorf("jwt may be invalid: %s", jwt)
-	}
-	payload := jwtSplit[1]
-
-	payloadBytes, err := base64.RawStdEncoding.DecodeString(payload)
-	if err != nil {
-		return false, fmt.Errorf("failed to decode jwt: %v", err.Error())
+//ValidateRemoteK8sJwt validates aaainst a remote k8s JWT at API server.
+func ValidateRemoteK8sJwt(targetJWT, jwtPolicy string, mc *kubecontroller.Multicluster) ([]string, error) {
+	tokenReview := &k8sauth.TokenReview{}
+	if jwtPolicy == jwt.JWTPolicyThirdPartyJWT {
+		tokenReview.APIVersion = "authentication.k8s.io/v1"
+		tokenReview.Kind = "TokenReview"
```

Comment:
nit: Using native k8s api, the kind and apiversion is unneccessary

Reply:
thanks, Fixed~